### PR TITLE
collections: add can_be_viewed_by method

### PR DESCRIPTION
**Pull request by SamiHiltunen**
- NEW Adds can_be_viewed_by method to Collection for checking
  whether the user has a permission to view the collection or
  not.
- Improves code quality compliance.

Signed-off-by: Sami Hiltunen sami.mikael.hiltunen@cern.ch

### CODE REVIEW

File: invenio_collections/models.py [hunk 1]
```diff
@@ -360,6 +367,14 @@ def sort_methods(self):
 
         return BsrMETHOD.query.order_by(BsrMETHOD.name).all()
 
+    def can_be_viewed_by(self, user_info):
+        """Check if the user has sufficient privileges to view the document."""
```
Comment (jirikuncar) [inline invenio_collections/models.py]: `s/document/collection/`

**Comment by SamiHiltunen** [comment]
Fixed the doctoring referring to document instead of collection.

File: invenio_collections/models.py [hunk 2]
```diff
@@ -360,6 +367,14 @@ def sort_methods(self):
 
         return BsrMETHOD.query.order_by(BsrMETHOD.name).all()
 
+    def can_be_viewed_by(self, user_info):
+        """Check if the user has sufficient rights to view the collection."""
+        if not self.is_restricted():
```
Comment (jirikuncar) [inline invenio_collections/models.py]: `is_restricted` should be a property not method

File: invenio_collections/models.py [hunk 3]
```diff
@@ -360,6 +367,14 @@ def sort_methods(self):
...
+            return True
+        (auth_code, auth_msg) = acc_authorize_action(
```
Comment (jirikuncar) [inline invenio_collections/models.py]: Please note that caching should be implemented as calling `acc_authorize_action` is pretty "expensive". Have a look at `precached_permitted_restricted_collections` key in `user_info` first.

**Comment by jirikuncar** [comment]
@SamiHiltunen please introduce simple tests for new method. Thanks

File: invenio_collections/models.py (hunk 2 above)
Comment (jirikuncar) [inline invenio_collections/models.py]: @SamiHiltunen ping!

**Comment by lnielsen** [comment]
Closing PR as it's been stale for a long time.

**Pull request closed without merging**